firefox-code-search/background.js

```diff
@@ -0,0 +1,83 @@
+const SRC_URL = "http://searchfox.org/mozilla-central";
+const PATH_PREFIX = "path:";
+
+chrome.omnibox.setDefaultSuggestion({
+  description: "Search the firefox codebase (e.g. `hello world` | `path:omnibox.js hello world`"
+});
+
+chrome.omnibox.onInputEntered.addListener((text, disposition) => {
+  let url = text;
+  if (!text.startsWith(SRC_URL)) {
+    url = `${SRC_URL}/search?q=${text}+-path%3Aobj-&redirect=false`
+  }
+  chrome.tabs.query({active: true, currentWindow: true}, tabs => {
+    chrome.tabs.update(tabs[0].id, {url});
+  });
```

Comment:
We've been trying to use `browser` and promises in the examples rather than callbacks.